### DISCUSSION TITLE
Examples - setting LORAWAN_JOIN_EUI, APP_KEY etc

### DISCUSSION THREAD

**Discussion by dmeehan1968** [Lora-net/LoRaMac-node#1009] [Q&A]
It appears that in the case of the soft-se secure element (and others), that there are hardcoded values for the joinEUI, appKey and nwkKey.

The docs seem to suggest using the LR1110 secure element which can be pre-provisioned with appropriate values, but this was not my use case.

I determined that it was possible to set these values prior to the initial `LmHandlerJoin()` call, such as:

```
// define as appropriate
#define APPEUI	{ 0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF, }
#define APPKEY	{ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  }
...
// main(), after LmHandlerInit() and prior to LmHandlerJoin()
SecureElementSetJoinEui((uint8_t [])SD01L_APPEUI);
SecureElementSetKey(APP_KEY, (uint8_t [])SD01L_APPKEY);
SecureElementSetKey(NWK_KEY, (uint8_t [])SD01L_APPKEY);
```

Alternatively, the sources should be modified to allow conditional definition of LORAWAN_JOIN_EUI, APP_KEY and NWK_KEY via preprocessor directives. eg.

```
// se-identity.h
#ifndef LORAWAN_JOIN_EUI
#define LORAWAN_JOIN_EUI { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, }
#endif

// also for other keys
```

Documentation seems poor in this regard, leaving the user to determine the preconfigured keys or understand which API calls should be used, and where to use them.

**Comment by mluis1** [comment]
Please refer to the discussion at following link https://github.com/Lora-net/LoRaMac-node/discussions/984 for further details on this subject.

In future release we will consider updating the LR1110 secure-element handling in order to ease it usage.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.
